docs/how-to-guides/format-api.md

```diff
@@ -0,0 +1,199 @@
+-   JavaScript setup for building and enqueuing
+
+The [complete format-api example]() is available that you can use as a reference for your setup.
+
+## Step-by-step guide
+
+The guide will refer to `src/index.js` as the JavaScript file where the changes are made. After each step, running `npm run build` creates `build/index.js` that is then loaded on the post editor screen.
+
+### Step 1: Register a new format
+
+The first step is to register the new format, add `src/index.js` with the following:
+
+```js
+import { registerFormatType } from '@wordpress/rich-text';
+
+registerFormatType( 'my-custom-format/sample-output', {
+	title: 'Sample output',
+	tagName: 'samp',
+	className: null,
+} );
+```
+
+The list of available format types is maintained in the `core/rich-text` store. You can query the store to check that your custom format is now available.
+
+Run this code in your browser's console to confirm:
+
+```js
+wp.data.select( 'core/rich-text' ).getFormatTypes();
+```
+
+It'll return an array containing the format types, including your own.
+
+### Step 2: Add a button to the toolbar
+
+With the format available, the next step is to add a button to the UI by registering a component for the edit property.
+
+Using the `RichTextToolbarButton` component, update `src/index.js`:
+
+```js
+import { registerFormatType } from '@wordpress/rich-text';
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+
+const MyCustomButton = ( props ) => {
+	return (
+		<RichTextToolbarButton
+			icon="editor-code"
+			title="Sample output"
+			onClick={ () => {
+				console.log( 'toggle format' );
+			} }
+		/>
+	);
+};
+
+registerFormatType( 'my-custom-format/sample-output', {
+	title: 'Sample output',
+	tagName: 'samp',
+	className: null,
+	edit: MyCustomButton,
+} );
+```
+
+Let's check that everything is working as expected. Build and reload and then select a text block. Confirm the new button was added to the format toolbar.
+
+![Toolbar with custom button](https://developer.wordpress.org/files/2021/12/format-api-toolbar.png)
+
+Click the button and check the console.log for the "toggle format" message.
+
+If you do not see the button or message, double check you are building and loading the JavScript properly; and check the console.log for any errors.
+
+### Step 3: Apply a format when clicked
+
+Next is to update the button to apply a format when clicked.
+
+For our example, the `<samp>` tag format is binary - either a text selection has the tag or not, so we can use the `toggleFormat` method from the RichText package.
+
+Update `src/index.js` changing the `onClick` action:
+
+```js
+import { registerFormatType, toggleFormat } from '@wordpress/rich-text';
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+
+const MyCustomButton = ( props ) => {
+	return (
+		<RichTextToolbarButton
+			icon="editor-code"
+			title="Sample output"
+			onClick={ () => {
+				props.onChange(
+					toggleFormat( props.value, {
```

Comment:
Wondering if we should use destructuring for props? My guess is that is what would be seen in production code.

Reply:
Switched here and the gutenberg-examples repo. 👍